tools/shared.py

```diff
@@ -3128,6 +3128,9 @@ def legalize_sig(sig):
 
   @staticmethod
   def is_legal_sig(sig):
+    # with BigInt support all sigs are legal since we can use i64s.
+    if Settings.WASM_BIGINT:
```

Comment:
Should legalize_sig be changed to `return sig` if WASM_BIGINT? That would make this still correct, and handle any other places we'd want the legalized signature

Reply:
Looks like this is not necessary, since the only callsite is already checking LEGALIZE_JS_FFI before calling.

Replace with `assert not Settings.WASM_BIGINT`?

Reply:
I think @jgravelle-google is right, since there is another call site to there, so this code path is actually taken with bigints. I missed this in testing since the existing tests happen to not check it carefully enough. Fixed + added a test now, thanks!

Reply:
But where is the other callsite?   I only see one.

Reply:
`legalize_sig` is called from `is_legal_sig` and also `make_invoke`. The latter path was not handled properly and we emitted slightly incorrect invoke code.

Reply:
`is_legal_sig` is looks like its only called from `emscripten.py:create_fp_accessors` 

So can't `is_legal_sig` just do `assert not Settings.WASM_BIGINT` since calling this when `WASM_BIGINT` should never happen.

Reply:
Ah, so there are two issues here.

The function @jgravelle-google and I were talking about is `legalize_sig`. That's not the function this discussion is tacked on to, it's the function @jgravelle-google mentioned we might be missing (and we were).

You are right about `is_legal_sig`, but I think instead of an assertion it's easy to just do the right behavior as the patch already does? Seems clearer, and ready if we need it later.

Reply:
Oh wait, no, `is_legal_sig` is called from `emscripten.py` there, yeah. So we can't assert, we have to do what the patch already does.

Reply:
Oh heh, no, that was wrong, ignore the last comment.